From f46c706fb86ad6aed08e959fb6b5da422d7a11bb Mon Sep 17 00:00:00 2001
From: Shawn Ferry <sferry@gmail.com>
Date: Wed, 16 Aug 2017 16:27:30 -0400
Subject: [PATCH 2/3] (MODULES-5494) puppetlabs-mysql : default version for
 Solaris should be 5.7

Mysql 5.5 has been EOLd and will be removed from Solaris. The new
default version should be 5.7.

If mysql is installed or mediated that version will be used instead.

change submitted to upstream
---
 lib/facter/mysql_solaris.rb            | 4 ++--
 spec/unit/facter/mysql_solaris_spec.rb | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/facter/mysql_solaris.rb b/lib/facter/mysql_solaris.rb
index c8f2e02..a0c5712 100644
--- a/lib/facter/mysql_solaris.rb
+++ b/lib/facter/mysql_solaris.rb
@@ -5,10 +5,10 @@ Facter.add('mysql_solaris') do
     ver = Facter::Util::Resolution.exec('pkg mediator -H mysql').split[2]
 
     # Above exec doesn't raise an error if the command returns non-zero results
-    ver = ver.nil? ? '5.5' : ver
+    ver = ver.nil? ? '5.7' : ver
   rescue
     # Just in case this raises an error in the future
-    ver = '5.5'
+    ver = '5.7'
   end
   setcode do
     {
diff --git a/spec/unit/facter/mysql_solaris_spec.rb b/spec/unit/facter/mysql_solaris_spec.rb
index 811d810..a41196e 100644
--- a/spec/unit/facter/mysql_solaris_spec.rb
+++ b/spec/unit/facter/mysql_solaris_spec.rb
@@ -47,17 +47,17 @@ describe Facter::Util::Fact do
       end
       context 'major_dot_minor' do
         it {
-          expect(Facter.fact(:mysql_solaris).value['major_dot_minor']).to eq('5.5')
+          expect(Facter.fact(:mysql_solaris).value['major_dot_minor']).to eq('5.7')
         }
       end
       context 'basedir' do
         it {
-          expect(Facter.fact(:mysql_solaris).value['basedir']).to eq('/usr/mysql/5.5')
+          expect(Facter.fact(:mysql_solaris).value['basedir']).to eq('/usr/mysql/5.7')
         }
       end
       context 'major_minor' do
         it {
-          expect(Facter.fact(:mysql_solaris).value['major_minor']).to eq('55')
+          expect(Facter.fact(:mysql_solaris).value['major_minor']).to eq('57')
         }
       end
     end
-- 
2.14.1

